hs_file_types/models/raster.py

```diff
@@ -209,6 +211,43 @@ def get_xml(self, pretty_print=True):
         return CoreMetaData.XML_HEADER + '\n' + etree.tostring(RDF_ROOT, encoding='UTF-8',
                                                                pretty_print=pretty_print).decode()
 
+    def get_preview_data_url(self, resource, folder_path):
+        """Generate a GeoServer layer preview link."""
+
+        if resource.raccess.public is True:
+            geoserver_url = settings.HS_GEOSERVER
+            resource_id = resource.short_id
+            layer_id = '.'.join('/'.join(folder_path.split('/')[2:]).split('.')[:-1])
+
+            for k, v in settings.HS_GEOSERVER_ESCAPE.items():
+                layer_id = layer_id.replace(k, v)
+
+            layer_id = quote(f'HS-{resource_id}:{layer_id}')
+
+            extent = quote(','.join((
+                str(self.spatial_coverage.value['westlimit']),
+                str(self.spatial_coverage.value['southlimit']),
+                str(self.spatial_coverage.value['eastlimit']),
+                str(self.spatial_coverage.value['northlimit']),
+            )))
+
+            layer_srs = quote(self.spatial_coverage.value['projection'][-9:])
+
+            preview_data_url = (
+                f'{geoserver_url}/HS-{resource_id}/wms'
+                f'?service=WMS&version=1.1&request=GetMap'
+                f'&layers={layer_id}'
+                f'&bbox={extent}'
+                f'&width=800&height=500'
+                f'&srs={layer_srs}'
+                f'&format=application/openlayers'
+            )
+
+        else:
+            preview_data_url = None
+
+        return preview_data_url
+
```

Comment:
I see a lot of duplicate code in get_preview_data_url() in raster.py and geofeature.py. Suggest to move the common code into an utility function so it can be called by both functions in raster.py and geofeature.py.

Reply:
5e2905f: Moved duplicate code to a utility function.